frappe/integrations/saml2.py

```diff
@@ -0,0 +1,125 @@
+    frappe.local.response["type"] = "redirect"
+    frappe.local.response["location"] = redirect_url
+
+@frappe.whitelist(allow_guest=True)
+def acs():
+    try:
+        # Handle the SAML response
+        post_data = dict(frappe.request.form.copy())
+        query_data = dict(frappe.request.args.copy())
+        request_data = {
+            "http_host": frappe.local.request.host,
+            "server_port": frappe.local.request.environ.get("SERVER_PORT"),
+            "script_name": frappe.local.request.environ.get("PATH_INFO"),
+            "post_data": post_data,
+            "https": "on" if frappe.local.request.scheme == "https" else "off",
+        }
+
+        saml_key = frappe.get_doc("Saml Login Key", query_data.get("provider"))
+        if not saml_key:
+            frappe.respond_as_web_page(_("SAML Login Failed"), _("SAML configuration missing"), http_status_code=500)
+            return
+
+        # Create the SAML settings dictionary
+        saml_settings = {
+            "sp": {
+                "entityId": saml_key.sp_entity_id,
+                "assertionConsumerService": {
+                    "url": frappe.utils.get_url("/api/method/frappe.integrations.saml2.acs"),
+                    "binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST"
+                },
+                "signatureAlgorithm": "RSA_SHA256",
+            },
+            "idp": {
+                "entityId": saml_key.idp_entity_id,
+                "singleSignOnService": {
+                    "url": saml_key.idp_sso_url,
+                    "binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"
+                },
+                "x509cert": saml_key.idp_x509cert,
+            }
+        }
+
+        client = OneLogin_Saml2_Auth(request_data, saml_settings)
+        client.process_response()
+
+        errors = client.get_errors()
+        if errors:
+            frappe.respond_as_web_page(_("SAML Login Failed"), _("Invalid SAML response: ") + ", ".join(errors), http_status_code=403)
+            return
+
+        if not client.is_authenticated():
+            frappe.respond_as_web_page(_("SAML Login Failed"), _("User not authenticated"), http_status_code=403)
+            return
+
+        # Extract user data
+        user_email = client.get_nameid()
+        if not user_email:
+            frappe.respond_as_web_page(_("SAML Login Failed"), _("Email not found in SAML response"), http_status_code=403)
+            return
+
+        # Check if the user exists
+        user = frappe.db.get_value("User", {"email": user_email})
+        if not user:
+            frappe.respond_as_web_page(_("SAML Login Failed"), _("User not found"), http_status_code=403)
+            return
+
+        # Log the user in
+        frappe.local.login_manager.user = user
+        frappe.local.login_manager.post_login()
+        frappe.db.commit()
+
+        # Redirect to workspace
+        frappe.local.response["type"] = "redirect"
```

Comment:
check if there is a way to redirect to use "redirect-to" param passed on `/login?redirect-to=<url>` if not set then redirect to `/me` instead of app.

expected to work like social login key.